MetaMask/metamask-extension
# Add privacy setting to disable external name sources

## **Description**

Add a toggle to the `Security & privacy` settings in the `Flask` build to disable the use of any `NameProvider` which uses an external source.

This disables the `EtherscanNameProvider`, `LensNameProvider`, and `TokenNameProvider` from making any callouts and therefore prevents any proposed names being displayed in the `NameDetails` component, except for any values provided by installed snaps.

## **Manual Testing Steps**

1. Go to test dApp.
2. Connect.
3. Switch to mainnet.
4. Copy [this URL](https://metamask.github.io/test-dapp/request?method=eth_signTypedData_v4&params=%5B%22CURRENT_ACCOUNT_ADDRESS%22,%7B%22domain%22:%7B%22chainId%22:1,%22name%22:%22Petnames%20Test%22,%22verifyingContract%22:%220xdac17f958d2ee523a2206206994597c13d831ec7%22,%22version%22:%221%22%7D,%22message%22:%7B%22scenarios%22:%5B%7B%22sources%22:%5B%22Etherscan%22,%22Token%22%5D,%22addresses%22:%5B%220xdac17f958d2ee523a2206206994597c13d831ec7%22,%220xB8c77482e45F1F44dE1745F52C74426C631bDD52%22,%220xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48%22,%220xae7ab96520de3a18e5e111b5eaab095312d7fe84%22%5D%7D,%7B%22sources%22:%5B%22ENS%22,%22Lens%20Protocol%22%5D,%22addresses%22:%5B%220xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045%22%5D%7D%5D%7D,%22primaryType%22:%22Petnames%20Test%22,%22types%22:%7B%22EIP712Domain%22:%5B%7B%22name%22:%22name%22,%22type%22:%22string%22%7D,%7B%22name%22:%22version%22,%22type%22:%22string%22%7D,%7B%22name%22:%22chainId%22,%22type%22:%22uint256%22%7D,%7B%22name%22:%22verifyingContract%22,%22type%22:%22address%22%7D%5D,%22Petnames%20Test%22:%5B%7B%22name%22:%22scenarios%22,%22type%22:%22Scenario%5B%5D%22%7D%5D,%22Scenario%22:%5B%7B%22name%22:%22sources%22,%22type%22:%22string%5B%5D%22%7D,%7B%22name%22:%22addresses%22,%22type%22:%22address%5B%5D%22%7D%5D%7D%7D%5D) and replace `%CURRENT_ACCOUNT_ADDRESS%` with your current account address.
5. This will load a V4 signature request in the test dapp including addresses which will should all have proposed names. 
6. Select the last address and verify there is a `Lens Protocol` and `ENS` proposed name.
7. Go to `Settings` > `Security & privacy` > `Suggest address names` and disable it.
8. Wait 2 minutes.
9. Go back to the above URL and select the last address.
10. Ensure only the ENS proposed name is shown.

## **Screenshots**

<img width="353" alt="petnames_external_sources_toggle" src="https://github.com/MetaMask/metamask-extension/assets/19745142/79c323be-9c94-4d6a-9de4-53ff39856600">

## **Pre-merge Author Checklist**


## **Pre-merge reviewer checklist**
